### ISSUE TITLE
OpenCL 3.0 Reference Pages

### ISSUE THREAD

**Issue by bashbaug** [KhronosGroup/OpenCL-Docs#665]
This issue is to track tasks and progress to update the OpenCL reference pages for OpenCL 3.0.

Task list (will grow):

* [ ] Update any of the static pages (including the main index page) from OpenCL 2.2 to 3.0.
* [x] Update the reference page links from the 2.2 specs to the 3.0 specs (mostly a script change?).
* [x] At least add stub pages for the missing links to fix the 404 errors, assuming the we can't refactor the spec to easily generate them.
* [ ] At least add stub pages for any new extensions (list TBD).
* [ ] Perhaps add pages for any new OpenCL 3.0 concepts (e.g. OpenCL C features, list TBD).

Some related issues (mostly for the 2.2 reference pages):

* https://github.com/KhronosGroup/OpenCL-Docs/issues/70
* https://github.com/KhronosGroup/OpenCL-Docs/issues/246
* https://github.com/KhronosGroup/OpenCL-Registry/issues/94

**Comment by oddhack** [comment]
@bashabaug #671 is my WIP branch for this. I have put a snapshot under https://www.khronos.org/registry/OpenCL/sdk/3.0/docs/man/html/ for review but this is just the first step.

@outofcontrol if you could run the link-checker over https://www.khronos.org/registry/OpenCL/sdk/3.0/docs/man/html/ that would help chase down problems. Note that these files are not part of the registry repo, they're just layered on top of the clone on kaylee to be on what will eventually be the right path. They are not wired into any of the site navigation and shouldn't be until the MR is ready for merge.

**Comment by outofcontrol** [comment]
No 404s or other anomalies.

**Comment by oddhack** [comment]
@bashbaug this is in a relatively good state. Looks like there are no dead links, refpages have been updated to link and refer by name to the 3.0 specs, content is extracted from current API / C language specs. The main remaining issue is updating static content and extension refpages. The extensions are themselves asciidoc markup now, so it should be relatively simple to collate them together with the pages extracted from the specs - I'll look at that. The static pages (under man/static/) need a close look from people very familiar with the 2.2 -> 3.0 changes to make sure those are reflected. With those two steps complete, I think we could merge the branch and formally publish 3.0 refpages.

The "New OpenCL concepts" bullet item could either be content in the API / C specs that we annotate as refpages so it can be extracted, or additional static pages.

**Comment by bashbaug** [comment]
Agreed, PR #671 looks good and I think we should merge it as-is.  Thanks!

I found one minor issue with `clCreateImageWithProperties` but we could fix it as part of the detailed 2.2 -> 3.0 upgrade.

How hard would it be to create a reference page version of the "OpenCL 3.0 Backwards Compatibility" appendix?  That's the biggest single upgrade I think we'd want to make, and the rest would be checking that no new APIs or other minor details are missed.

https://www.khronos.org/registry/OpenCL/specs/3.0-unified/html/OpenCL_API.html#opencl-3.0-backwards-compatibility

**Comment by oddhack** [comment]
> Agreed, PR #671 looks good and I think we should merge it as-is. Thanks!
> 
> I found one minor issue with `clCreateImageWithProperties` but we could fix it as part of the detailed 2.2 -> 3.0 upgrade.
> 
> How hard would it be to create a reference page version of the "OpenCL 3.0 Backwards Compatibility" appendix? That's the biggest single upgrade I think we'd want to make, and the rest would be checking that no new APIs or other minor details are missed.

It is hard to fit it into the existing refpage framework, which is aimed at extracting short bits of related text for a single API. I'd have to make some script changes to allow extracting the entire appendix.

Would it work to create a short static refpage that describes this in a general sense, then deep links to the spec appendix for the details?
